File: app/src/main/java/com/ariefzuhri/myspecificnotification/rest/ApiService.java
package com.ariefzuhri.myspecificnotification.rest;

import com.ariefzuhri.myspecificnotification.model.Sender;
import com.ariefzuhri.myspecificnotification.response.MyResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Headers;
import retrofit2.http.POST;

import static com.ariefzuhri.myspecificnotification.BuildConfig.SERVER_KEY;

public interface ApiService {

    @Headers(
            {
                    "Content-Type:application/json",
                    "Authorization:key=" + SERVER_KEY
            }
    )

    @POST("fcm/send")
    Call<MyResponse> sendNotification(@Body Sender body);
}

File: app/src/main/java/com/ariefzuhri/myspecificnotification/data/network/ApiService.kt
package com.ariefzuhri.myspecificnotification.data.network

import com.ariefzuhri.myspecificnotification.data.response.FcmResponse
import retrofit2.http.POST
import com.ariefzuhri.myspecificnotification.BuildConfig
import com.ariefzuhri.myspecificnotification.data.model.Message
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Headers

interface ApiService {

    @Headers(
        "Content-Type:application/json",
        "Authorization:key= ${BuildConfig.FIREBASE_SERVER_KEY}"
    )

    @POST("fcm/send")
    fun sendNotification(@Body body: Message): Call<FcmResponse>
}